Update user-facing error and CLI messages to align with the style guidelines

edmorley opened the pull request:
The CX team has very kindly offered to review all user facing error and CLI messages.

This PR originally contained a script to dump all messages to a markdown file for easier review, which has been removed now that the review and messaging changes have taken place.

In addition to updating several of the messages, this PR adds `project.toml` files to several of the existing test fixtures, which allows them to now be run against the CLI. (Previously those testcases would only work when run against individual function validation steps as part of the unit tests, however it's useful to be able to fall back to running the testcases directly with the CLI too.)

GUS-W-12442542.
GUS-W-12477495.

edmorley commented:
cc @jshackell-sfdc

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: Could not read project.toml: UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte
````
```suggestion
Function failed validation: Couldn't read project.toml: UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte.
```

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81738]
````
```suggestion
INFO:     Started server process [81738].
```

jshackell-sfdc commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81738]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: Could not read project.toml: UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte
```
```suggestion
ERROR:    RuntimeError: Unable to load function: Couldn't read project.toml: UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte.
```

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+In the context of the production deploy self-check:
+
+```term
+Function failed validation: project.toml is missing the required '[com.salesforce]' table.
````
```suggestion
Function failed validation: The project.toml file is missing the required '[com.salesforce]' table.
```

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81740]
````
```suggestion
INFO:     Started server process [81740].
```

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81754]
````
```suggestion
INFO:     Started server process [81754].
```

jshackell-sfdc commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://localhost:8080 (Press CTRL+C to quit)
+invocationId=00DJS0000000123ABC-0baf07e3d0f724a9b944165a86e51f89 level=error msg="Function return value cannot be serialized: TypeError: Type is not JSON serializable: set"
```
```suggestion
invocationId=00DJS0000000123ABC-0baf07e3d0f724a9b944165a86e51f89 level=error msg="Function return value can't be serialized: TypeError: Type is not JSON serializable: set."
```

edmorley replied:
Same as https://github.com/heroku/sf-functions-python/pull/78/files#r1093776099 for the trailing full stop.

jshackell-sfdc commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://localhost:8080 (Press CTRL+C to quit)
+level=error msg="Could not parse CloudEvent: Data payload is not valid JSON: invalid literal: line 1 column 1 (char 0)"
```
```suggestion
level=error msg="Couldn't parse CloudEvent: Data payload isn't valid JSON: invalid literal: line 1 column 1 (char 0)."
```

edmorley commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+Salesforce Functions Python Runtime
+
+options:
+  -h, --help            show this help message and exit
```
Unfortunately the `--help` messages are auto-generated by the Python stdlib's `argparse` module, so we cannot capitalise these unless I manually implement the module's built-in functionality, which I'd prefer not to do (especially given most users won't be interacting with the `sf-functions-python` CLI directly, and only the `sf` CLI).

jshackell-sfdc replied:
Gotcha, that's fine!

edmorley commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+  <project-path>  The directory that contains the function
+
+options:
+  -h, --help      show this help message and exit
```
Same as https://github.com/heroku/sf-functions-python/pull/78/files#r1093758684

edmorley commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+INFO:     Started server process [81699]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://localhost:8080 (Press CTRL+C to quit)
```
Same as https://github.com/heroku/sf-functions-python/pull/78/files#r1093760852

edmorley commented on user-facing-messages.md:
```diff
@@ -0,0 +1,432 @@
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81736]
+INFO:     Waiting for application startup.
+ERROR:    RuntimeError: Unable to load function: project.toml is not valid TOML: Expected '=' after a key in a key/value pair (at line 1, column 4)
```
For this and the next one, the changes to the start of the error message look great, however I'm less sure about the trailing full stop at the end, only because we don't generate the error message that's appended - so the library might include some that already have a trailing full stop, which would mean a double full stop.

jshackell-sfdc replied:
makes sense!  so ignore it.  thx for explaining :)

jshackell-sfdc commented on user-facing-messages.md:
````diff
@@ -0,0 +1,432 @@
+
+```term
+Starting sf-functions-python v0.4.0 in single process mode.
+INFO:     Started server process [81699]
````
I wondered that, but edited it anyway, just in case.  So no worries!

edmorley commented:
@jshackell-sfdc I've addressed all of the review comments except those where I commented otherwise, in:
https://github.com/heroku/sf-functions-python/pull/78/commits/af241266d7e8d63105a32cb80dc05a6d26277538

However, I noticed that the rendered output you reviewed actually had the wrong error message for a few of the testcases - it had lots of duplicates of the "Didn't find a project.toml` error variant, rather than the intended message for that testcase. The reason for this was that in the unit tests these testcases exercise the individual validation functions (so don't need to pass all the other checks), but when run via the CLI the other checks error out before we even get to the error message they were meant to test.

I've fixed those testcases now, so they can be run via the CLI too - and then regenerated the rendered output. Would you be able to review these messages too? GitHub allows commenting on commits too, so to save having to sift through the whole file again, I would comment directly on here:
https://github.com/heroku/sf-functions-python/pull/78/commits/34eb3e1367088650b1e2943261166a7e16653832#diff-49f7f0472fc1a7580b6d3bec7bf1e86d8f71f7b3ca0962a2de9785911371948f

jshackell-sfdc commented:
@edmorley : Just one suggested change (x 4) here: https://github.com/heroku/sf-functions-python/commit/34eb3e1367088650b1e2943261166a7e16653832#diff-49f7f0472fc1a7580b6d3bec7bf1e86d8f71f7b3ca0962a2de9785911371948f

Otherwise, looks great!

jshackell-sfdc commented:
And thanks for doing all this regeneration then porting backwards, then regeneration again... Must be a real drag for you!

edmorley commented:
Thank you for the second review. 

It's not actually too bad for this PR - I was able to just run the script again (it only takes 10 seconds or so).
